daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2562,3 +2562,35 @@ func NewDeviceSet(root string, doInit bool, options []string, uidMaps, gidMaps [
 
 	return devices, nil
 }
+
+// ReloadPool reloads the pool.
+func (devices *DeviceSet) ReloadPool(dataFileName, metadataFileName string) error {
+	datafile, err := os.OpenFile(dataFileName, os.O_RDWR, 0)
+	if datafile == nil {
+		return err
+	}
+	defer datafile.Close()
+
+	metadatafile, err := os.OpenFile(metadataFileName, os.O_RDWR, 0)
+	if metadatafile == nil {
+		return err
+	}
+	defer metadatafile.Close()
+
+	// Suspend the pool
+	if err := devicemapper.SuspendDevice(devices.getPoolName()); err != nil {
+		return fmt.Errorf("Unable to suspend pool: %s", err)
+	}
+
+	// Reload with the new block sizes
+	if err := devicemapper.ReloadPool(devices.getPoolName(), datafile, metadatafile, devices.thinpBlockSize); err != nil {
+		return fmt.Errorf("Unable to reload pool: %s", err)
```

Comment:
What happens if user is running docker with dm.blocksize=X which is different from default block size. Then this logic is broken.